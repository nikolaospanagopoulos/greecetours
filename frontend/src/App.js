import Header from "./components/Header.js/Header";
import { BrowserRouter as Router,Route } from "react-router-dom";
import Footer from "./components/Footer.js/Footer";
import ProductsPage from "./Pages/ProductsPage/ProductsPage";
import TourPage from './Pages/TourPage/TourPage';
import CartPage from './Pages/CartPage/CartPage'
import LoginPage from './Pages/LoginPage/LoginPage'
import HomePage from './Pages/HomePage/HomePage.jsx'
import "./App.css";

function App() {
  return (
    <Router>
      <Header />
      <main>
      <Route path='/login'  component={LoginPage}/>
      <Route path='/cart/:id?'  component={CartPage}/>
      <Route path='/tours/tour/:id'  component={TourPage}/>
      <Route path='/tours/:sort?' exact  component={ProductsPage}/>
      <Route path='/' exact component={HomePage}/>
      </main>
      <Footer />
    </Router>
  );
}

export default App;
