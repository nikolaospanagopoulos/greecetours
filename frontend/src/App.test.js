
import {shallow} from 'enzyme'
import React from 'react'
import App from './App'



it('expect to render App Component',() => {
    expect(shallow(<App/>).length).toEqual(1)
})

it('expect to render App Component',() => {
    expect(shallow(<App/>)).toMatchSnapshot()
})