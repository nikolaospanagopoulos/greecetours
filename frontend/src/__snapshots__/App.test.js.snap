// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render App Component 1`] = `
<BrowserRouter>
  <Header />
  <main>
    <Route
      component={[Function]}
      path="/order/:id"
    />
    <Route
      component={[Function]}
      path="/finalize"
    />
    <Route
      component={[Function]}
      path="/tourbuy"
    />
    <Route
      component={[Function]}
      path="/services"
    />
    <Route
      component={[Function]}
      path="/login"
    />
    <Route
      component={[Function]}
      path="/about"
    />
    <Route
      component={[Function]}
      path="/payment"
    />
    <Route
      component={[Function]}
      path="/register"
    />
    <Route
      component={[Function]}
      path="/profile"
    />
    <Route
      component={[Function]}
      path="/cart/:id?"
    />
    <Route
      component={[Function]}
      path="/tours/tour/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/tours/:sort?"
    />
    <Route
      component={[Function]}
      path="/admin/userlist"
    />
    <Route
      component={[Function]}
      path="/admin/user/:id/edit"
    />
    <Route
      component={[Function]}
      path="/admin/tourlist"
    />
    <Route
      component={[Function]}
      path="/admin/tour/:id/edit"
    />
    <Route
      component={[Function]}
      path="/admin/orderlist"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
  </main>
  <Footer />
</BrowserRouter>
`;
